iroh-net/src/relay/client.rs

```diff
@@ -244,25 +249,92 @@ impl<W: AsyncWrite + Unpin + Send + 'static> ClientWriter<W> {
     }
 }
 
-/// The Builder returns a [`Client`] starts a [`ClientWriter`] run task.
+/// The Builder returns a [`Client`] and a started [`ClientWriter`] run task.
 pub struct ClientBuilder {
     secret_key: SecretKey,
-    reader: RelayReader,
-    writer: FramedWrite<MaybeTlsStreamWriter, DerpCodec>,
-    local_addr: SocketAddr,
+    reader: RelayConnReader,
+    writer: RelayConnWriter,
+    local_addr: Option<SocketAddr>,
+}
+
+pub(crate) enum RelayConnReader {
+    Derp(FramedRead<MaybeTlsStreamReader, DerpCodec>),
+    Ws(SplitStream<WebSocketStream>),
+}
+
+pub(crate) enum RelayConnWriter {
+    Derp(FramedWrite<MaybeTlsStreamWriter, DerpCodec>),
+    Ws(SplitSink<WebSocketStream, tokio_tungstenite_wasm::Message>),
+}
+
+fn tung_wasm_to_io_err(e: tokio_tungstenite_wasm::Error) -> std::io::Error {
+    std::io::Error::new(std::io::ErrorKind::Other, e.to_string())
```

Comment:
`tokio_tungstenite_wasm::Error` already has a `Io(io::Error)` variant that would be good to avoid re-wrapping as `Other`